controller/area_blackbox_test.go

```diff
@@ -59,190 +57,190 @@ func (rest *TestAreaREST) UnSecuredController() (*goa.Service, *AreaController)
-}
-
-func (rest *TestAreaREST) TestFailCreateChildAreaWithInvalidsParent() {
-	// given
-	createChildAreaPayload := newCreateChildAreaPayload("TestFailCreateChildAreaWithInvalidsParent")
-	svc, ctrl := rest.SecuredController()
-	// when/then
-	test.CreateChildAreaNotFound(rest.T(), svc.Context, svc, ctrl, uuid.NewV4().String(), createChildAreaPayload)
-}
-
-func (rest *TestAreaREST) TestFailCreateChildAreaNotAuthorized() {
-	// given
-	_, parentArea := createSpaceAndArea(rest.T(), rest.db)
-	parentID := parentArea.ID
-	createChildAreaPayload := newCreateChildAreaPayload("TestFailCreateChildAreaNotAuthorized")
-	svc, ctrl := rest.UnSecuredController()
-	// when/then
-	test.CreateChildAreaUnauthorized(rest.T(), svc.Context, svc, ctrl, parentID.String(), createChildAreaPayload)
-}
-
-func (rest *TestAreaREST) TestFailValidationAreaNameLength() {
-	// given
-	ci := newCreateChildAreaPayload(testsupport.TestOversizedNameObj)
-
-	err := ci.Validate()
-	// Validate payload function returns an error
-	assert.NotNil(rest.T(), err)
-	assert.Contains(rest.T(), err.Error(), "length of type.name must be less than or equal to 63")
-}
-
-func (rest *TestAreaREST) TestFailValidationAreaNameStartWith() {
-	// given
-	ci := newCreateChildAreaPayload("_TestSuccessCreateChildArea")
-
-	err := ci.Validate()
-	// Validate payload function returns an error
-	assert.NotNil(rest.T(), err)
-	assert.Contains(rest.T(), err.Error(), "type.name must match the regexp")
-}
-
-func (rest *TestAreaREST) TestFailShowAreaNotFound() {
-	// given
-	svc, ctrl := rest.SecuredController()
-	// when/then
-	test.ShowAreaNotFound(rest.T(), svc.Context, svc, ctrl, uuid.NewV4().String(), nil, nil)
-}
-
-func (rest *TestAreaREST) TestShowAreaOK() {
-	// given
-	_, a := createSpaceAndArea(rest.T(), rest.db)
-	svc, ctrl := rest.SecuredController()
-	// when
-	res, _ := test.ShowAreaOK(rest.T(), svc.Context, svc, ctrl, a.ID.String(), nil, nil)
-	//then
-	assertResponseHeaders(rest.T(), res)
-}
+func (rest *TestAreaREST) TestCreateChildArea() {
+	rest.T().Run("Success", func(t *testing.T) {
+		t.Run("OK", func(t *testing.T) {
+			// given
+			fxt := tf.NewTestFixture(t, rest.DB, tf.Spaces(1), tf.Areas(1))
+			sp := fxt.Spaces[0]
+			parentArea := fxt.Areas[0]
+			parentID := parentArea.ID
+			ca := newCreateChildAreaPayload("TestSuccessCreateChildArea")
+			owner, err := rest.db.Identities().Load(context.Background(), sp.OwnerID)
+			require.NoError(t, err)
+			svc, ctrl := rest.SecuredControllerWithIdentity(owner)
+			// when
+			_, created := test.CreateChildAreaCreated(t, svc.Context, svc, ctrl, parentID.String(), ca)
+			// then
+			assert.Equal(t, *ca.Data.Attributes.Name, *created.Data.Attributes.Name)
+			fmt.Println(*created.Data.Relationships.Parent.Data.ID)
+			assert.Equal(t, parentID.String(), *created.Data.Relationships.Parent.Data.ID)
+
+			// try creating child area with different identity: should fail
```

Comment:
Please remove creation of identity here and just add `tf.Identities(2)` to your test fixture recipe above. Then you can use `otherIdentity := *fxt.Identities[1]` as an alternative identity. Please note, that unless told otherwise the test fixture always uses the first (index 0) resource if there's a dependency on it and if unless being told otherwise. Therefore if you use index 1, you get an identity that is not used for anything else.

Reply:
Yes. This makes it so easy to write tests.